File: server/Core/Models/ApiResponse.cs
namespace Core.Models
{
    public class ApiResponse
    {
        public bool IsSuccess { get; set; }
        public string Message { get; set; }
        public object? Data { get; set; }

        public static ApiResponse Failure(string message)
        {
            return new ApiResponse
            {
                IsSuccess = false,
                Message = message
            };
        }

        public static ApiResponse Failure(string message, object data)
        {
            return new ApiResponse
            {
                IsSuccess = false,
                Message = message,
                Data = data
            };
        }

        public static ApiResponse Failure()
        {
            return new ApiResponse
            {
                IsSuccess = false,
                Message = null,
                Data = null
            };
        }

        public static ApiResponse Success()
        {
            return new ApiResponse
            {
                IsSuccess = true,
                Message = null,
                Data = null
            };
        }

        public static ApiResponse Success(string message)
        {
            return new ApiResponse
            {
                Data = null,
                Message = message,
                IsSuccess = true
            };
        }

        public static ApiResponse Success(string message, object data)
        {
            return new ApiResponse
            {
                Data = data,
                Message = message,
                IsSuccess = true
            };
        }

        public static ApiResponse Success(object data)
        {
            return new ApiResponse
            {
                Data = data,
                Message = string.Empty,
                IsSuccess = true
            };
        }
    }
}

File: mobile/app/src/main/java/com/n7/quizbattle/shared/models/ApiResponse.java
package com.n7.quizbattle.shared.models;

import com.google.gson.annotations.SerializedName;

public class ApiResponse<T> {
    @SerializedName("isSuccess")
    private boolean isSuccess;

    @SerializedName("message")
    private String message;

    @SerializedName("data")
    private T data;

    // Required for Gson
    public ApiResponse() {
    }

    public ApiResponse(boolean isSuccess, String message, T data) {
        this.isSuccess = isSuccess;
        this.message = message;
        this.data = data;
    }

    public boolean isSuccess() {
        return isSuccess;
    }

    public void setSuccess(boolean success) {
        isSuccess = success;
    }

    public String getMessage() {
        return message;
    }

    public void setMessage(String message) {
        this.message = message;
    }

    public T getData() {
        return data;
    }

    public void setData(T data) {
        this.data = data;
    }
}
